discovery_engine_core/core/src/document.rs

```diff
@@ -149,18 +150,25 @@ impl TryFrom<Article> for NewsResource {
 
 /// Indicates user's "sentiment" towards the document,
 /// essentially if the user "liked" or "disliked" the document.
-#[derive(Clone, Copy, Debug, Derivative, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, Derivative, PartialEq, Serialize_repr, Deserialize_repr)]
 #[derivative(Default)]
+#[repr(u8)]
 pub enum UserReaction {
     /// No reaction from the user.
     #[derivative(Default)]
-    Neutral,
+    Neutral = 0,
 
     /// The user is interested.
-    Positive,
+    Positive = 1,
 
     /// The user is not interested.
-    Negative,
+    Negative = 2,
+}
+
+#[derive(Debug, DisplayDoc, Error)]
+/// Received an unsupported user reaction, int repr: {reaction}
+pub struct UnsupportedUserReaction {
```

Comment:
This seems unused.

Reply:
Right, I had a `TryFrom` implementation before I change to writing to the pointer directly.